pimd/pim_upstream.c

```diff
@@ -207,9 +215,7 @@ struct pim_upstream *pim_upstream_del(struct pim_instance *pim,
 	if (pim_up_mlag_is_local(up))
 		pim_mlag_up_local_del(pim, up);
 
-	THREAD_OFF(up->t_ka_timer);
-	THREAD_OFF(up->t_rs_timer);
-	THREAD_OFF(up->t_msdp_reg_timer);
+	pim_upstream_timers_stop_helper(up);
```

Comment:
In the function pim_upstream_timers_stop_helper, it is doing THREAD_OFF(up->t_join_timer);
Again in the join_timer_stop(up), it is doing THREAD_OFF(up->t_join_timer);

Can you please look into this.

Reply:
I am really confused by this question.  This is for when a shutdown is happening for a particular upstream.  Going through and ensuring that all timers are stopped( especially ones that hold pointers ) is good form.  What are you expecting to happen and why?